### ISSUE TITLE
Create a shared library for server startup checks

### ISSUE THREAD

**Issue by garethbowen** [medic/medic-webapp#4683]
We have a bunch of code in api/server.js and sentinel/server.js which checks certain prerequisites (eg: node version, logging env.NODE_OPTIONS) before starting. This is duplicated but also gets out of sync if new checks are added to one and not the other. Create a shared library with all the checks that both services should run.

**Comment by browndav** [comment]
Great idea, thanks.

**Comment by newtewt** [comment]
@garethbowen  @vimemo  

I'm running node 6.14.3 on my VM. When I run node server.js in the api folder I'm getting an error about promisify not being a function because we require ```migrations = require('./src/migrations') ``` and node is trying to load that file. We don't get to the server checks first and the user won't see our requirements first. 

Where as if I run node server.js in the sentinel folder I get the expected error `Node version 6.14.3 is not supported, minimum 8.0.0`

**Comment by newtewt** [comment]
Another question, there are two checks for environment variables happening. Would we want to maintain both?

https://github.com/medic/medic-webapp/blob/87f5dba0ba5a3f280c61b62a06fba6154a285607/api/src/db-pouch.js#L36-L40

and  

https://github.com/medic/medic-webapp/blob/87f5dba0ba5a3f280c61b62a06fba6154a285607/shared-libs/server-checks/src/checks.js#L21-L37

**Comment by garethbowen** [comment]
@newtewt Good points!

I think we could run the checks before we `require` code so we can pick up the insufficient node version before node fails to compile.

As for the db-pouch.js that's mostly to catch incorrectly executed unit testing. We could clean this up so it just includes that last line, because you're right, if you're running the full stack it'll fail before db-pouch runs.

@vimemo What do you think?

**Comment by vimemo** [comment]
Sounds good.

**Comment by newtewt** [comment]
@vimemo Hey Victor the PR is still open any update here?

**Comment by vimemo** [comment]
@newtewt, forgot about this ticket. Submitting another update here shortly. thank you.

**Comment by vimemo** [comment]
@garethbowen, @newtewt :
This has been merged to master and 3.0.x.
Thank you.

**Comment by newtewt** [comment]
@garethbowen  @vimemo  

The suggested change below has db-pouch being loaded first. That has a check for the COUCH_URL being set. This means if the COUCH_URL env variable isn't set we'll see `If you are running unit tests use UNIT_TEST_ENV=1 in your environment.` error first. If that's expected then I'll close this. Just seems a little off.

```
const db = require('./src/db-pouch'),
      serverChecks = require('@shared-libs/server-checks');

serverChecks.check(db.serverUrl).then(() => {
  const config = require('./src/config'),
        migrations = require('./src/migrations'),
        ddocExtraction = require('./src/ddoc-extraction'),
        translations = require('./src/translations'),
        serverUtils = require('./src/server-utils'),
        apiPort = process.env.API_PORT || 5988;

  Promise.resolve()
    .then(() => console.log('Extracting ddoc…'))
// etc
});
```

**Comment by vimemo** [comment]
@newtewt, I'll revert to include those logs with the right message. That should be better. thank you.

**Comment by newtewt** [comment]
@vimemo  any update on the revert?

**Comment by vimemo** [comment]
@newtewt , it has been merged in 3.0 and master; last week.

**Comment by newtewt** [comment]
@vimemo  thanks I didn't see it for some reason. Closing this ticket.
